test/e2e/scheduling/priorities.go

```diff
@@ -339,6 +339,94 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		framework.ExpectNoError(err)
 		framework.ExpectEqual(tolePod.Spec.NodeName, nodeName)
 	})
+
+	ginkgo.Context("PodTopologySpread Scoring", func() {
+		var nodeNames []string
+		topologyKey := "kubernetes.io/e2e-pts-score"
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("Trying to get 2 available nodes which can run pod")
+			nodeNames = Get2NodesThatCanRunPod(f)
+			ginkgo.By(fmt.Sprintf("Apply dedicated topologyKey %v for this test on the 2 nodes.", topologyKey))
+			for _, nodeName := range nodeNames {
+				framework.AddOrUpdateLabelOnNode(cs, nodeName, topologyKey, nodeName)
+			}
+		})
+		ginkgo.AfterEach(func() {
+			for _, nodeName := range nodeNames {
+				framework.RemoveLabelOffNode(cs, nodeName, topologyKey)
+			}
+		})
+
+		ginkgo.It("validates pod should be preferably scheduled to node which makes the matching pods more evenly distributed", func() {
+			var nodes []v1.Node
+			for _, nodeName := range nodeNames {
+				node, err := cs.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				nodes = append(nodes, *node)
+			}
+
+			// Make the nodes have balanced cpu,mem usage.
+			err := createBalancedPodForNodes(f, cs, ns, nodes, podRequestedResource, 0.5)
```

Comment:
I know we need this so that the "node resource" Score plugin doesn't influence, but this is going to cause flakes because a pod created by a previous test may get deleted after this function is called, and so breaking the "balanced" assumption. I don't have a better suggestion how to fix this though. 

Giving PodTopologySpread a higher priority should make this less important.

Reply:
Exactly like what I mentioned in https://github.com/kubernetes/kubernetes/issues/88174#issuecomment-586470458. And this is why I hesitated to add this score test in e2e - you can't adjust the default weight (which is 1) for a given e2e env.